src/Tools/ExternalAccess/Razor/RazorAnalyzerAssemblyResolver.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Composition;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    [Export(typeof(IAnalyzerAssemblyResolver)), Shared]
+    internal class RazorAnalyzerAssemblyResolver : IAnalyzerAssemblyResolver
+    {
+        internal static Func<AssemblyName, Assembly?>? AssemblyResolver;
```

Comment:
>I am assuming the MEF composition will still work, I could be wrong 

That's the kicker here. In OOP the razor and roslyn MEF catalogs don't have any visibility to each other. The usual pattern would be to declare the concrete type in Roslyn, along with an external access interface that Razor provides the impl of via MEF, but because of the above restriction we can't.

I frankly *hate* this static, it's an ugly hack, but it is at least limited to the contract between razor and Roslyn. 

More than happy to make the EA contract more resilient to changes if we can.

Reply:
I wonder if we might be ok in this instance though, this particular bit of EA should only be used by the razor tooling, which (I think?) should always be sim-shipping with roslyn. That might not be true for VSCode/devkit though, will need to find that out.

Reply:
> In the past, we have had difficulty shipping updates to EA due to need to load newer Roslyn into old versions of Razor. e.g. #72129.

The Razor EA problems I've been aware of have all been for the EA layer that shipped with the compiler. That lesson we learned from that is an EA layer that can see the compiler _is_ a compiler DLL. It must be shipped next to the compiler in all deployments and loaded from that location. That was very problematic because it meant that every host of the compiler (workspaces, VS, VS code, etc ...) had to know to deploy + load that DLL.

My understanding is that this EA assembly is in the VS / VS Code layer. That is a much more limited set of hosts and they have already solved the problem of making sure the EA assembly is deployed correctly.

> I frankly hate this static, it's an ugly hack, but it is at least limited to the contract between razor and Roslyn.

My biggest worry is that there is a load order dependency here. Essentially this `static` must be set before the assembly resolve happens. If it happens the other way then we just silently fail here.

At the least can we add some guards to tell us if this happened in the wrong order? Basically if the `static` is set _after_ the resolve event for razor occurs that should be a red flag. At least a `Debug.Assert` failure.